### ISSUE TITLE
Finally block not executed on exception in async function

### ISSUE THREAD

**Issue by pelchats** [google/closure-compiler#2572]
The following code gets compiled in such a way that the finally block won't get executed if an exception is thrown in the try block:
```javascript
async function f() {
  try {
    await g();
    console.log('IN TRY BLOCK');
  } finally {
    console.log('IN FINALLY BLOCK');
  }
}
```
Adding an empty catch block resolves the problem.

Looking at the state machine in the generated code, one can see the code is wrong (case 1 doesn't fallthrough because an exception is thrown):
```javascript
        switch(c) {
          //...
          case 6:
            try {
              console.log("IN TRY BLOCK");
              e = 3;
              c = 2;
              break;
            } catch (k) {
              d = k;
              c = 1;
              break;
            }
          case 1:
            throw h = d, c = -1, h;
          case 2:
            console.log("IN FINALLY BLOCK");
            c = e;
            break;
```
By comparison, adding the catch block yields the following code (case 1 does fallthrough):

```javascript
        switch(d) {
          //...
          case 5:
            try {
              console.log("IN TRY BLOCK");
              h = 3;
              d = 2;
              break;
            } catch (b) {
              d = 1;
              break;
            }
          case 1:
            h = 3;
          case 2:
            console.log("IN FINALLY BLOCK");
            d = h;
            break;
```

http://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520SIMPLE_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%2540formatting%2520pretty_print%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250Aasync%2520function%2520f()%2520%257B%250A%2520%2520try%2520%257B%250A%2520%2520%2520%2520await%2520g()%253B%250A%2520%2520%2520%2520console.log('IN%2520TRY%2520BLOCK')%253B%250A%2520%2520%252F%252F%257D%2520catch%2520(e)%2520%257B%250A%2520%2520%257D%2520finally%2520%257B%250A%2520%2520%2520%2520console.log('IN%2520FINALLY%2520BLOCK')%253B%250A%2520%2520%257D%250A%257D

**Comment by brad4d** [comment]
I expect to fix this as part of fixing #2504

**Comment by brad4d** [comment]
I believe this was fixed along with #2504
